.github/workflows/main.yaml

```
@@ -0,0 +1,73 @@
+name: Main CI
+
+on: [push]
+
+jobs:
+  lint:
+    runs-on: ubuntu-latest
+    timeout-minutes: 10
+    strategy:
+      matrix:
+        python-version:
+          - '3.12'
+    steps:
+    - uses: actions/checkout@v4
```

```suggestion
    - uses: actions/checkout@v4
      with:
        submodules: true
```

As per https://github.com/actions/checkout

Applies to the below again.

---

Good to know we can do this, but I agree with @davidlehn that we should get this merged before #182 and can revisit this if/when #182 makes it in.